framework/sandstone_ssl_rand.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include "sandstone_ssl.h"
+#include "sandstone_ssl_rand.h"
+
+static OSSL_PROVIDER *s_prov;
+
+/* Random context management */
+static void *sandstone_rand_newctx(void *ctx, void *parent,
+                            const OSSL_DISPATCH *parent_calls)
+{
+    // Allocate context
+    int *rand_st = reinterpret_cast<int*>(malloc(sizeof(*rand_st)));
+    if (rand_st != NULL)
+        *rand_st = EVP_RAND_STATE_UNINITIALISED;
+
+    return rand_st;
+}
+
+static void sandstone_rand_freectx(void *ctx)
+{
+     free(ctx);
+}
+
+/* Random number generator functions */
+static int sandstone_rand_instantiate(void *ctx, unsigned int strength, int prediction_resistance,
+                               const unsigned char *pstr, size_t pstr_len,
+                               const OSSL_PARAM params[])
+{
+    ctx = reinterpret_cast<void *>(EVP_RAND_STATE_READY);
+    return 1;
+}
+
+static int sandstone_rand_uninstantiate(void *ctx)
+{
+    ctx = reinterpret_cast<void *>(EVP_RAND_STATE_UNINITIALISED);
+    return 1;
+}
+
+static int sandstone_rand_generate(void *ctx, unsigned char *out, size_t outlen,
+                            unsigned int strength, int prediction_resistance,
+                            const unsigned char *addin, size_t addin_len)
+{
+    memset_random(out, outlen);
+
+    return 1;
+}
+
+// TODO: Do something
```

Comment:
We don't support reseeding. You can pretend to OpenSSL that we are a true RNG (that is, not a DPRNG).

Reply:
It's ok, it is not mandatory to support it. So I removed it.